app/src/main/java/at/roteskreuz/stopcorona/screens/debug/exposure_notifications/DebugExposureNotificationsFragment.kt

```diff
@@ -114,15 +138,15 @@ class DebugExposureNotificationsFragment : BaseFragment(R.layout.debug_contact_t
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
         when (requestCode) {
-            DebugExposureNotificationsViewModel.DebugAction.REGISTER_WITH_FRAMEWORK.requestCode() -> {
+            ExposureNotificationRepository.DebugAction.REGISTER_WITH_FRAMEWORK.requestCode() -> {
```

Comment:
Request codes are always the responsibility of the View layer. The caller should set the correct request code and also the caller should check if the request code matches.